plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorOutlinePage.java

```diff
@@ -148,7 +153,7 @@ public void updateChildCount(@NotNull Object element, int currentChildCount) {
                 } else if (element instanceof OutlineNode node) {
                     treeViewer.setChildCount(element, node.getChildrenCount());
                 } else {
-                    throw new UnsupportedOperationException();
+                    treeViewer.setChildCount(element, rootNodes.size());
```

Comment:
Why intentionally supporting root object model (which is EditorInput) being out of sync with the editor instead of just listen for the corresponding event of it being changed and update the outline model to be in sync?

Reply:
Is this resolved?

Reply:
I'm still not sure that we should do this change, but ok, let's try.